content/general/concepts/middleware/middleware.md

```diff
@@ -0,0 +1,66 @@
+---
+Title: 'Middleware'
+Description: 'Middleware is software that connects data, APIs, software tools, and other applications in order to bring a complete application to the end user.'
+Subjects:
+  - 'Computer Science'
+  - 'Web Development'
+Tags:
+  - 'APIs'
+  - 'Database'
+  - 'Middleware'
+CatalogContent:
+  - 'learn-sql'
+  - 'paths/back-end-engineer-career-path'
+---
+
+Middleware is software that connects applications together. The term "middleware" comes from the fact that this software sits between the front-end, or client-side, of an application and the back-end, or server-side, where the resources are being requested.
+
+Data, APIs, software tools, and other applications are connected by middleware in order to bring a complete or bundled application to the end user. There are many types of middleware software depending on the use cases. Some of the most commonly used middleware is listed below.
+
+## Types of Middleware
+
+### Message Oriented Middleware
+
+A message oriented middleware, or MOM, allows applications to exchange messages and communicate between one another. This middleware offers ease of use but can be costly at larger scales.
+
+#### Examples
+
+- RabbitMQ
+
+RabbitMQ is a messaging broker platform used by applications to send and recieve messages.
+
+- IBM MQ
+
+IBM MQ is another example of message oriented middleware. Some of the core components include messages, message queues and queue manager.
+
+- Amazon SNS
+
+Amazon simple notification service is a way to manage messaging between applications and users.
+
+### Database middleware
+
+This type of middleware gives access to databases and provides points of interaction between a variety of databases. This middleware improves efficiency however it can be language specific which can create limitations.
+
+#### Examples
+
+- ODBC
+
+Open Database Connectivity is an API used for accessing database management systems.
```

Comment:
Can we try placing these in a toggle. When we click on the toggle, it should expand to show the additional info on the middleware.

```
<details>
  <summary>ODBC</summary>
  Open Database Connectivity is an API used for accessing database management systems.
</details>
```

Reply:
> Can we try placing these in a toggle. When we click on the toggle, it should expand to show the additional info on the middleware.
> 
> ```
> <details>
>   <summary>ODBC</summary>
>   Open Database Connectivity is an API used for accessing database management systems.
> </details>
> ```

Should I do it for all the examples ? or just this one ?